### ISSUE TITLE
DOCS Suggestion: Create follow along tutorials like the Pastebin tutorial for all the major examples from the examples folder

### ISSUE THREAD

**Issue by saikatdas0790** [SergioBenitez/Rocket#2435]
**Existing Functionality**

A clear and concise description of existing functionality and why it is insufficient. Examples:

To provide some context, I've mostly been a TS/JS developer who's used Express style backend frameworks to create backend REST APIs. 
I'm coming to Rocket after having gone through the Rust book, solving the Rustlings course and hand type all the examples from Rust by example. 
So, although I'm starting to feel fairly comfortable with the Rust language in general, I'm still struggling with advanced things like lifetimes, interior mutability and reference counting as well as threading and thread safety.
But being able to type out the follow along examples like the Pastebin tutorial lets me additively understand concepts and features that are gradually introduced. I believe this would be really helpful for new-ish users of the Rocket framework

**Suggested Changes**

Follow along tutorials like the Pastebin tutorial for all the major examples in the repo

**Alternatives Considered**

I've tried to type out the examples from the repo but it's quite confusing since they are not introducing a single feature/concept at a time additively to the project and hence quite difficult to trace the logical flow when faced with the entire project

**Additional Context**

I can understand that this would be a significantly large addition to the docs but I believe every example added would bring more clarity to the framework and Rust in general. This is because it would also showcase Rust's powers in a real life framework in different scenarios instead of just some synthetic examples.

**Comment by SergioBenitez** [comment]
I agree that Rocket would benefit from more official tutorials like the Pastebin one. I'd love to see suggestions for such a tutorial, and ideally an implementation. Here are a few ideas:

  * Turn the `chat` example into a tutorial.
  * Turn the `todo` example into a tutorial.
  * Expand the `databases` example into a `blog` tutorial.

I'm converting this into a discussion to continue there.
